src/sentry/api/endpoints/sentry_app_avatar.py

```diff
@@ -0,0 +1,30 @@
+from rest_framework.response import Response
+
+from sentry import features
+from sentry.api.bases import SentryAppBaseEndpoint
+from sentry.api.bases.avatar import AvatarMixin
+from sentry.models import SentryAppAvatar
+
+
+class SentryAppAvatarEndpoint(AvatarMixin, SentryAppBaseEndpoint):
+    object_type = "sentry_app"
+    model = SentryAppAvatar
+
+    def get(self, request, **kwargs):
+        sentry_app = kwargs.get("sentry_app", None)
+
+        if features.has("organizations:sentry-app-logo-upload", sentry_app.owner):
+            return super().get(request, access=request.access, **kwargs)
+        else:
+            return Response({"detail": ["You do not have that feature enabled"]}, status=400)
```

Comment:
No need for the else here if you're returning in the if.

I would probably reverse this actually to make the feature check a ['guard clause'](https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html)

```suggestion
        if not features.has("organizations:sentry-app-logo-upload", sentry_app.owner):
            return Response({"detail": ["You do not have that feature enabled"]}, status=400)

        return super().get(request, access=request.access, **kwargs)
```